File: app/lib/magazzino/models/piano_model.dart
// ignore_for_file: non_constant_identifier_names

import 'package:ows_test_v2/generale/models/abstract_model.dart';
import 'package:ows_test_v2/magazzino/models/montante_model.dart';

class PianoModel extends AbstractModel {
  int id_piano;
  String? cod_piano;
  String? desc_piano;
  String? cod_alias;
  int? id_magazzino;
  int? id_zona;
  int? id_corridoio;
  int? id_montante;
  DateTime? data_ins;
  DateTime? data_agg;

  MontanteModel? montante;

  PianoModel({
    required this.id_piano,
    this.cod_piano,
    this.desc_piano,
    this.cod_alias,
    this.id_magazzino,
    this.id_zona,
    this.id_corridoio,
    this.id_montante,
    this.data_ins,
    this.data_agg,
    this.montante,
  });

  @override
  Map<String, dynamic>? toJson() {
    return <String, dynamic>{
      "id_piano": id_piano,
      "cod_piano": cod_piano,
      "desc_piano": desc_piano,
      "cod_alias": cod_alias,
      "id_magazzino": id_magazzino,
      "id_zona": id_zona,
      "id_corridoio": id_corridoio,
      "id_montante": id_montante,
      "data_ins": data_ins,
      "data_agg": data_agg,
      "montante": montante?.toJson(),
    };
  }

  factory PianoModel.fromJson(Map<String, dynamic> json) {
    return PianoModel(
      id_piano: json["id_piano"],
      cod_piano: json["cod_piano"],
      desc_piano: json["desc_piano"],
      cod_alias: json["cod_alias"],
      id_magazzino: json["id_magazzino"],
      id_zona: json["id_zona"],
      id_corridoio: json["id_corridoio"],
      id_montante: json["id_montante"],
      data_ins: DateTime.tryParse(json["data_ins"] ?? ''),
      data_agg: DateTime.tryParse(json["data_agg"] ?? ''),
      montante: json["montante"] != null ? MontanteModel.fromJson(json["montante"]) : null,
    );
  }
}

File: web/Magazzino/Models/PianoModel.cs
using System.Data.SqlClient;
using web.Generale;

namespace web.Magazzino.Models
{
    
    public class PianoModel:BasicModel
    {
        public int Id_piano { get; set; }
        public string? Cod_piano { get; set; }
        public string? Desc_piano { get; set; }
        public string? Cod_alias { get; set; }
        public int? Id_magazzino { get; set; }
        public int? Id_zona { get; set; }
        public int? Id_corridoio { get; set; }
        public int? Id_montante { get; set; }

        public MontanteModel? Montante { get; set; }

        public PianoModel()
        {
            NomeTabella = "piano";
            GestioneAzienda = false;
            GestioneLogUtenti = false;
        }

       

        public override void FromReader(SqlDataReader reader)
        {
            FromReaderCommon(reader);
            Id_piano  = GetIntNonNullable(reader, nameof(Id_piano));
            Cod_piano = GetString(reader, nameof(Cod_piano));
            Desc_piano = GetString(reader, nameof(Desc_piano));
            Cod_alias = GetString(reader, nameof(Cod_alias));
            Id_magazzino = GetInt(reader, nameof(Id_magazzino));
            Id_zona = GetInt(reader, nameof(Id_zona));
            Id_corridoio = GetInt(reader, nameof(Id_corridoio));
            Id_montante = GetInt(reader, nameof(Id_montante));

            Montante = new MontanteModel();
            Montante.FromReader(reader);
        }
    }
}
